deeptime/decomposition/_score.py

```diff
@@ -0,0 +1,105 @@
+        score is computed for one realization of self and `test_model`. Estimation
+        of the average cross-validation score and partitioning of data into test and
+        training part is not performed by this method.
+
+        If `covariances_test` is None, this method computes the VAMP score for the model
+        contained in self.
+
+    dim : int, optional, default=None
+        Artificially restrict the scoring to the top `dim` slowest processes.
+
+    epsilon : float, default=1e-6
+
+
+    Returns
+    -------
+    score : float
+        If `test_model` is not None, returns the cross-validation VAMP score between
+        self and `test_model`. Otherwise return the selected VAMP-score of self.
+
+    Notes
+    -----
+    The VAMP-:math:`r` and VAMP-E scores are computed according to :cite:`vampscore-wu2020variational`,
+    Equation (33) and Equation (30), respectively.
+
+    References
+    ----------
+    .. bibliography:: /references.bib
+        :style: unsrt
+        :filter: docname in docnames
+        :keyprefix: vampscore-
+    """
+    if dim is not None:
+        dim = min(koopman_model.koopman_matrix.shape[0], dim)
+    if isinstance(r, str):
+        r = r.lower()
+        assert r == 'e', "only VAMP-E supported, otherwise give as float >= 1"
+    else:
+        assert isinstance(r, numbers.Number) and r >= 1, "score only for r >= 1 or r = \"E\""
+    if covariances_test is None:
+        cov_test = koopman_model.cov
+    else:
+        cov_test = covariances_test
+    assert koopman_model.cov.data_mean_removed == cov_test.data_mean_removed, \
+        "Covariances must be consistent with respect to the data"
+    if koopman_model.cov.cov_00.shape != cov_test.cov_00.shape:
+        raise ValueError("Shape mismatch, the covariances ")
+    if not is_sorted(koopman_model.singular_values, 'desc'):
+        sort_ix = np.argsort(koopman_model.singular_values)[::-1][:dim]  # indices to sort in descending order
+    else:
+        sort_ix = np.arange(koopman_model.singular_values.shape[0])[:dim]  # already sorted
+
+    U = koopman_model.instantaneous_coefficients[:, sort_ix]
+    V = koopman_model.timelagged_coefficients[:, sort_ix]
+
+    if r == 'e':
+        K = np.diag(koopman_model.singular_values[sort_ix])
+        # see https://arxiv.org/pdf/1707.04659.pdf eqn. (30)
+        score = np.trace(2.0 * np.linalg.multi_dot([K, U.T, cov_test.cov_0t, V])
+                         - np.linalg.multi_dot([K, U.T, cov_test.cov_00, U, K, V.T, cov_test.cov_tt, V]))
+    else:
+        # see https://arxiv.org/pdf/1707.04659.pdf eqn. (33)
+        A = np.atleast_2d(spd_inv_sqrt(U.T.dot(cov_test.cov_00).dot(U), epsilon=epsilon))
+        B = np.atleast_2d(U.T.dot(cov_test.cov_0t).dot(V))
+        C = np.atleast_2d(spd_inv_sqrt(V.T.dot(cov_test.cov_tt).dot(V), epsilon=epsilon))
+        ABC = np.linalg.multi_dot([A, B, C])
+        score = schatten_norm(ABC, r) ** r
+    if koopman_model.cov.data_mean_removed:
+        score += 1
```

Comment:
Depends on whether the mean of the data was removed during estimation of the covariance matrix. If it was removed, this also removes the constant function from the singular function space of the koopman operator (contributing an additional constant of '1'). 
This is mainly for consistency: If comparing the score from VAMP estimator (mean removed here) and MSM estimator (mean not removed), then this correction makes the scores compatible and comparable.

Reply:
Yes, I understood this rationale. I am just not sure if one wouldn't expect that the score is lower by -1 when estimating from mean-free data compared to non-whitened data. It should be easy to find in the documentation at least, otherwise people will start wondering why they get the same score for the mean-free data.

Reply:
I will add a remark to the method's documentation for now. To be honest I am not sure if we even should remove the mean, perhaps it would be cleaner to just remove the constant component out of the projection, some methods even artificially reintroduce the constant function into the koopman operator matrix.